audit/models.go

```diff
@@ -0,0 +1,111 @@
+package admin
+
+import (
+	"time"
+)
+
+// Request base request
+type Request struct {
+	// usually an email address, but can be something else
+	// for email see the limit here https://stackoverflow.com/a/574698
+	// for other like names or ids 254 is also enough
+	RequestedBy *string `json:"requestedBy,omitempty"  binding:"required,printascii,max=254"`
+	Reason      *string `json:"reason,omitempty"  binding:"required,printascii,max=512"`
+}
+
+// IChangeRequest general change request interface
+type IChangeRequest interface {
```

Comment:
`IChangeRequest` is not a good interface name.
How about `IDSetter` if it only has that one function?
Or `ChangeMaker`, `ChangeRequestMaker`?

Reply:
@viettd This is the first PR, in the next PR, will have a general function to generic the `bindXXXRequest` like this
``` go
// BindGeneralChangeRequest Bind general change request(Update/Delete)
func BindGeneralChangeRequest(c *gin.Context, ctxKey string, request IChangeRequest) (err error) {
	// try to get from context
	ctxReq, ok := c.Get(ctxKey)
	if ok {
		request, ok = ctxReq.(IChangeRequest)
		if ok {
			return
		}
	}

	// try to bind from request & set to context if ok
	if err = c.ShouldBindJSON(request); err != nil {
		err = errors.New(errors.BadRequest, err)
		return
	}
	var id uint64
	idParam := c.Param("id")
	id, err = strconv.ParseUint(idParam, 10, 64)
	if err != nil || id == 0 {
		err = errors.New(errors.BadRequest, fmt.Sprintf("invalid id [%s]", idParam))
		return
	}
	request.SetID(uint(id))
	c.Set(ctxKey, request)
	return
}
```
I think `IDSetter` or `xxxMaker` maybe not good to understand

Reply:
I see.I'm ok with this for now since I don't have a better suggestion.